src/lib.rs

```diff
@@ -274,6 +275,40 @@ impl UltraLogLog {
     pub fn get_state(&self) -> &[u8] {
         &self.state
     }
+    /// Serializes UltraLogLog to a file using bincode
+    /// the serde feature must be enabled
+    pub fn save(&self, path: &str) -> std::io::Result<()> {
+        // Convert the sketch to a bincode-encoded Vec<u8>
+        let encoded: Vec<u8> = bincode::serialize(&self)
+            .map_err(|e| std::io::Error::new(std::io::ErrorKind::Other, e))?;
+
+        // Write bytes to disk
+        std::fs::write(path, &encoded)
+    }
+
+    /// Loads an UltraLogLog from a bincode file
+    /// the serde feature must be enabled
+    pub fn load(path: &str) -> std::io::Result<Self> {
```

Comment:
The same as `save`. We can require the `bytes` as input, instead of reading it ourselves.